apps/server/src/controllers/userControllers/getUpcomingActivities.ts

```diff
@@ -31,6 +31,12 @@ export const getUpcomingUserActivities = async (
       },
     })
 
+    if (!userGroups) {
+      return res.status(HTTP_STATUS_CODE.NOT_FOUND).json({
+        message: 'user groups not found.',
+      })
+    }
+
```

Comment:
We don't need this, we just send an empty array in the case where there's no userGroup found... I believe this shouldn't error.